ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,338 @@
+"""
+Module responsible for translating g2p data into GA4GH native
+objects.
+"""
+
+import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+from urlparse import urlparse, urlunsplit
+import rdflib
+
+
+class G2PDataset:
+    """
+    an rdf object store
+    """
+
+    # keep the graph as a static variable
+    rdfGraph = None
+
+    def __init__(self, sources):
+
+        """
+        sources [{source,format}]
+        """
+
+        self._searchQuery = """
+            PREFIX OBAN: <http://purl.org/oban/>
+            PREFIX OBO: <http://purl.obolibrary.org/obo/>
+            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+            PREFIX faldo: <http://biohackathon.org/resource/faldo#>
+            PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+            SELECT %PROPERTIES%
+                WHERE {
+                    ?s    a OBAN:association .
+                    ?s  OBAN:association_has_subject ?l .
+                    ?l rdfs:label ?location_label  .
+                    %LOCATION%
+                    ?s  OBO:RO_has_environment  ?drug .
+                    ?drug  rdfs:label ?drug_label  .
+                    ?s  OBAN:association_has_object  ?d .
+                    ?d  rdfs:label ?disease_label  .
+                    ?d rdf:type ?disease .
+                    ?s  OBAN:association_has_object_property  ?evidence .
+                    OPTIONAL {  ?evidence  rdfs:label ?evidence_label } .
+              %FILTER%
+            }
+            """
+
+        if G2PDataset.rdfGraph is not None:
+            return
+
+        # initialize graph
+        G2PDataset.rdfGraph = rdflib.Graph()
+
+        # load with data
+        for source in sources:
+            if not source['format']:
+                G2PDataset.rdfGraph.parse(source['source'])
+            else:
+                G2PDataset.rdfGraph.parse(source['source'],
+                                          format=source['format'])
+        # TODO is this necessary?
+        self.associationsLength = 0
```

Comment:
We should keep all instance variables as private by default. Change to ``self._associationsLength = 0``.